src/data/multiset.lean

```diff
@@ -1975,6 +1975,48 @@ instance : semilattice_sup_bot (multiset α) :=
   bot_le := zero_le,
   ..multiset.lattice }
 
+theorem exists_smul_of_dvd_count (s : multiset α) {k : ℕ} :
+  (∀ (a : α), k ∣ multiset.count a s) → ∃ (u : multiset α), s = k • u :=
```

Comment:
I think that we need a lemma `s.to_finset.sum (λ a, s.count a • ([a] : multiset α)) = s` (or with `erase_dup` instead of `to_finset` if you want to have it in `data/multiset`). Then you can take `u = s.to_finset.sum (λ a, (s.count a / k) • ([a] : multiset α))` and apply `multiset.

Reply:
Thanks Yury! I'll give this a try.

Reply:
Hi @paulvanwamelen, just checking in on this PR. Need any help with it? It would be nice to get this, and your work on triples, in!

Reply:
Thanks Scott, I haven't abandoned this and am still working on it. I'll PR the pythagorean triples soon and then get back to this.